### PR TITLE
Allow TLS flags to be disabled

### LINKED ISSUE #1438: Disabling TLS in the collector keeps tls=true option
As reported by @kenfinnigan, setting the option `reporter.grpc.tls.enabled` to false on a CR that has it set to true (like in a CR on OpenShift), will cause the option to be set at both true and false.

### PR DESCRIPTION
**Pull request by rubenvp8510**
Fixes #1438 

Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

### CODE REVIEW

File: pkg/deployment/all_in_one.go [hunk 1]
```diff
@@ -63,15 +63,17 @@ func (a *AllInOne) Get() *appsv1.Deployment {
 
 	configmap.Update(a.jaeger, commonSpec, &options)
 	sampling.Update(a.jaeger, commonSpec, &options)
-	tls.Update(a.jaeger, commonSpec, &options)
-	ca.Update(a.jaeger, commonSpec)
-	ca.AddServiceCA(a.jaeger, commonSpec)
+	if len(util.FindItem("--collector.grpc.tls.enabled", options)) == 0 {
+		tls.Update(a.jaeger, commonSpec, &options)
```
Comment (jpkrohling) [inline pkg/deployment/all_in_one.go:67]: This doesn't feel right: if the option is set, then tls.Update will be skipped (and ca, and ca.AddServiceCA). Why did we have this as part of all reconciliation loops before? What happens when on the first reconciliation we did not have this parameter, causing the operator to add `--reporter.grpc.tls.enabled=true`, and the user then manually sets this to `false`?

**Review by jpkrohling — COMMENTED** [review]
Could you please confirm that you ran a manual test like the following?

- On OpenShift, create a simple CR. This should enable TLS by default
- Change the CR to disable TLS
- Check that the CR (and underlying Jaeger config) look sane and without extra options like paths to the certs and keys.

File: pkg/deployment/all_in_one.go (hunk 1 above)
Comment (rubenvp8510) [reply pkg/deployment/all_in_one.go:67]: Good point, I need to test this case, 

I ran manually tests only for the first case, I'll check what happens in the case of an update to the CR. but it seems like you're right.

**Comment by rubenvp8510** [comment]
I did some changes to only touch the flags in case of the TLS certificates, still creating CA on the reconciliation, regarding of the flags.

File: pkg/deployment/all_in_one.go (hunk 1 above)
Comment (jpkrohling) [reply pkg/deployment/all_in_one.go:69]: Did you test this case?

**Review by jpkrohling — COMMENTED** [review]
This looks good, but it's not still clear to me what happens when a collector is first provisioned with the TLS options, and then has the TLS options disabled. Did you test this case? Is it possible to simulate this with an e2e test?

File: pkg/deployment/all_in_one.go (hunk 1 above)
Comment (rubenvp8510) [reply pkg/deployment/all_in_one.go:69]: I tested it with the last changes I did, it seems to work fine.

**Comment by jpkrohling** [comment]
@rubenvp8510 could you please rebase this?

**Comment by rubenvp8510** [comment]
> This looks good, but it's not still clear to me what happens when a collector is first provisioned with the TLS options, and then has the TLS options disabled. Did you test this case? Is it possible to simulate this with an e2e test?

The TLS will be disabled, but all the volumes will be still there.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**